### PR TITLE
Fix newline problems

### PR DESCRIPTION
**Pull request by jocgir**
- Update dependencies
- Add new Sprig functions
- Properly handle the last newline when handling direct input to stdout
- Fix bug when the error is reported in another file 
- Fix static-check error (i.e. Errors should not start with capital letter)
- Fix run function to not pass context as shell argument
- Do not ignore args if exec args are not totally split eg. @run("ls -l". "nothing") vs @run("ls", "-l", "nothing")

### CODE REVIEW

File: template/extra_data.go [hunk 1]
```diff
@@ -313,10 +312,10 @@ func toQuotedTFVars(v interface{}) (string, error) {
...
 }
 
-func toXML(v interface{}) (string, error) {
-	output, err := xml.Marshal(v)
-	return string(output), err
-}
+// func toXML(v interface{}) (string, error) {
+// 	output, err := xml.Marshal(v)
+// 	return string(output), err
+// }
```
Comment (sbeginCoveo) [inline template/extra_data.go:318]: should we keep these comments?

File: template/extra_data.go [hunk 2]
```diff
@@ -461,9 +460,9 @@ func (t *Template) templateConverter(to marshaler, from unMarshaler, source inte
 	return
 }
 
-func (t *Template) xmlConverter(source interface{}, context ...interface{}) (interface{}, error) {
-	return t.templateConverter(xml.Marshal, xml.Unmarshal, source, context...)
-}
+// func (t *Template) xmlConverter(source interface{}, context ...interface{}) (interface{}, error) {
+// 	return t.templateConverter(xml.Marshal, xml.Unmarshal, source, context...)
+// }
```
Comment (sbeginCoveo) [inline template/extra_data.go:465]: idem

File: template/extra_math.go [hunk 3]
```diff
@@ -433,10 +433,10 @@ func gamma(a interface{}) (r interface{}, err error) {
...
 }
 
-func infinity(a interface{}) (r interface{}, err error) {
-	defer func() { err = trapError(err, recover()) }()
-	return simplify(math.Inf(toInt(a))), nil
-}
+// func infinity(a interface{}) (r interface{}, err error) {
+// 	defer func() { err = trapError(err, recover()) }()
+// 	return simplify(math.Inf(toInt(a))), nil
+// }
```
Comment (sbeginCoveo) [inline template/extra_math.go:439]: idem

File: template/extra_sprig.go [hunk 4]
```diff
@@ -209,7 +224,10 @@ var sprigFuncRef = map[string]struct {
 	"valuesSprig":    {group: sprigDict, description: "Is similar to 'keys', except it returns a new 'list' with all the values of the source 'dict' (only one dictionary is supported).", aliases: []string{"values"}},
 
 	// Lists functions
+	"all":          {group: sprigList, description: "Filters items out of a list.", aliases: []string{"without"}},
+	"any":          {group: sprigList, description: "Takes a list of values and returns true if any values are non-empty."},
```
Comment (sbeginCoveo) [inline template/extra_sprig.go:228]: will these new functions be documented?

File: template/template_error_handler.go [hunk 5]
```diff
@@ -39,6 +39,10 @@ func (t errorHandler) Handler(err error) (string, bool, error) {
 		// We remove the faulty line and continue the processing to get all errors at once
 		lines := strings.Split(t.Code, "\n")
 		faultyLine := toInt(matches[tagLine]) - 1
+		if actualLine := matches[tagActualLine]; actualLine != "" {
+			// The actual error occurred in another line
+			faultyLine = toInt(actualLine) - 1
+		}
```
Comment (sbeginCoveo) [inline template/template_error_handler.go:45]: :+1:

**Review by sbeginCoveo — APPROVED** [review]

File: template/extra_data.go (hunk 1 above)
Comment (jocgir) [reply template/extra_data.go:318]: I kept them because we may restore it once some problem are resolved.

File: template/extra_data.go (hunk 2 above)
Comment (jocgir) [reply template/extra_data.go:465]: Idem

File: template/extra_math.go (hunk 3 above)
Comment (jocgir) [reply template/extra_math.go:439]: Idem

**Pull request merged**